src/System.IO.Compression/tests/CompressionStreamUnitTests.Gzip.cs

```diff
@@ -32,6 +32,137 @@ public void Precancellation()
             }
         }
 
+        [Fact]
+        public void ConcatenatedGzipStreams()
+        {
+            using (MemoryStream concatStream = new MemoryStream())
+            {
+                using (var gz = new GZipStream(concatStream, CompressionLevel.NoCompression, true))
+                using (var sw = new StreamWriter(gz))
+                    sw.WriteLine("Stream 1");
+
+                using (var gz = new GZipStream(concatStream, CompressionLevel.NoCompression, true))
+                using (var sw = new StreamWriter(gz))
+                    sw.WriteLine("Stream 2");
+
+                new GZipStream(concatStream, CompressionLevel.NoCompression, true).Dispose();
+
+                concatStream.Seek(0, SeekOrigin.Begin);
+                using (var gz = new GZipStream(concatStream, CompressionMode.Decompress))
+                using (var sr = new StreamReader(gz))
+                {
+                    Assert.Equal("Stream 1", sr.ReadLine());
+                    Assert.Equal("Stream 2", sr.ReadLine());
+                    Assert.Empty(sr.ReadToEnd());
+                }
```

Comment:
Any test for a stream containing gzip data followed by normal uncompressed data, or do we not support that today?

Reply:
There is a test for that already that shows how we read past the end of the data stream to fill our buffer: https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/IO/Compression/CompressionStreamUnitTestBase.cs#L234

This PR doesn't affect that test.